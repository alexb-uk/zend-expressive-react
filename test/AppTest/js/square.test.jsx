import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";

import Square from "../../../src/App/js/components/square";

describe("<Square />", () => {
    it("renders <Square /> component", () => {
        const component = renderer.create(
            <Square value={1}/>
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("Player 1 move", () => {
        const wrapper = shallow(<Square/>);

        // Initial state
        expect(wrapper.text()).toEqual("");

        wrapper.simulate("click");
        expect(wrapper.text()).toEqual("X");
    });
});
