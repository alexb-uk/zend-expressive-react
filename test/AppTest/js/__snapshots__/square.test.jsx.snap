// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders <Square /> component 1`] = `
<button
  className="square"
  onClick={[Function]}
/>
`;
